### ISSUE TITLE
Circle Tool behaves strangely when control points are on a line

### ISSUE THREAD

**Issue by Marian-Braendle** [otfried/ipe#473]
![ellipse_tool_bug](https://github.com/otfried/ipe/assets/14273595/f668aa29-f2a2-49ac-91e9-025ef28f32a9)

## Steps to Reproduce

1. Create Circle
2. Drag both control point onto a line
3. Move first control point

## Context

- Ipe 7.2.27
- Ubuntu 22.04

**Comment by Marian-Braendle** [comment]
Addendum:
If you drag the control point of Step 3 far enough, the program will simply crash without any output.
It creates a segfault:
```
ipe[22291]: segfault at 10 ip 00007fdae193d021 sp 00007ffd7534c378 error 4 in libQt6Gui.so.6.4.3[7fdae1722000+658000] likely on CPU 0 (core 0, socket 0)
Code: 31 c0 48 85 d2 74 03 8b 42 18 c3 66 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 00 f3 0f 1e fa 48 8b 0e 48 63 d2 48 89 f8 48 8d 14 52 <48> 8b 71 10 48 8d 14 d6 f3 0f 6f 02 48 8b 52 10 0f 11 07 48 89 57
```

**Comment by Marian-Braendle** [comment]
I looked a bit into the implementation and I noticed that it should not really be allowed to drag the second control point onto a line with the first control point, as this should make the matrix for control point 2 singular:
https://github.com/otfried/ipe/blob/26485ee58aab21e56a19e9b9e64f97bb794d739c/src/ipe/lua/editpath.lua#L64-L85

However, `isSingular` only outputs `true` if the determinant is exactly 0: https://github.com/otfried/ipe/blob/26485ee58aab21e56a19e9b9e64f97bb794d739c/src/ipelua/ipeluageo.cpp#L288-L294

Presumably one should rather work with a threshold here?

**Comment by otfried** [comment]
Fixed in 7.2.28.
